### ISSUE TITLE
NullreferenceException for SwaggerRequestExample

### ISSUE THREAD

**Issue by vhe1** [mattfrear/Swashbuckle.Examples#7]
Hi!
This is my controller method:
`        [HttpPut]
        [Route(UriHelper.ClicksRoute)]
        [SwaggerRequestExample(typeof(Dictionary<int,int>), typeof(ClicksCountValueExampleProvider))]
        public async Task PutAsync([FromBody]Dictionary<int,int> clicksByOfferId)
            => await _clicksManager.SetClicksAsync(clicksByOfferId);`

this is my provider
`public class ClicksReturnValueExampleProvider : IExamplesProvider
{
    public object GetExamples()
    {
        return new Dictionary<int, int>
        {
            { 864469, 1438752 },
            { 864470, 1769866 }
        };
    }
}`

and my swagger config:
`c.OperationFilter<ExamplesOperationFilter>();`

Now, for another call in the same controller, the return value model stuff works perfectly. But for this put call here, I get an error:
> An error has occurred.","exceptionMessage":"Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.","exceptionType":"System.NullReferenceException","stackTrace":" bei Swashbuckle.Examples.ExamplesOperationFilter.SetRequestModelExamples(Operation operation, SchemaRegistry schemaRegistry, ApiDescription apiDescription)
 bei Swashbuckle.Examples.ExamplesOperationFilter.Apply(Operation operation, SchemaRegistry schemaRegistry, ApiDescription apiDescription)
 bei Swashbuckle.Swagger.SwaggerGenerator.CreateOperation(ApiDescription apiDesc, SchemaRegistry schemaRegistry)
 bei Swashbuckle.Swagger.SwaggerGenerator.CreatePathItem(IEnumerable`1 apiDescriptions, SchemaRegistry schemaRegistry)
 bei Swashbuckle.Swagger.SwaggerGenerator.<>c__DisplayClass7.<GetSwagger>b__4(IGrouping`2 group)
 bei System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
 bei Swashbuckle.Swagger.SwaggerGenerator.GetSwagger(String rootUrl, String apiVersion)

**Comment by vhe1** [comment]
In the examples package, method SetRequestModelExamples, the schema is pretty much empty:
![grafik](https://user-images.githubusercontent.com/12475356/28690512-c1efdac6-7319-11e7-81c8-3a51c4b4422a.png)
and then var parts = schema.@ref.Split('/'); fails.

I'm attaching the controller and provider.

[SampleFiles.zip](https://github.com/mattfrear/Swashbuckle.Examples/files/1181162/SampleFiles.zip)

**Comment by mattfrear** [comment]
The NullReferenceException was fixed on 09 August in commit bbd1fe46. However, even with that fix the SwaggerRequestExamples still doesn't work - I am looking into it.

**Comment by mattfrear** [comment]
Firstly, thank you for the detailed bug report.

The problem is your controller action method's parameter is a Dictionary<int, int>. Swashbuckle doesn't do anything with this. If you look at the underlying swagger.json, one of my working examples has this:
```
"parameters": [

    {
        "name": "personRequest",
        "in": "body",
        "required": true,
        "schema": {
            "$ref": "#/definitions/PersonRequest"
        }
    }

],
```
All my library does is look for #/definitions/PersonRequest at the end of the swagger document, and adds the example:
```
"definitions": {
    "PersonRequest": {
        "type": "object",
        "properties": {
            "Income": {
                "format": "double",
                "type": "number"
            }
        },
        "example": {
            "Title": "Mr",
            "Age": 24,
            "FirstName": "Dave"
        }
```
Now, your Dictionary<int, int> example has this:
```
"parameters": [
    {
        "name": "personRequest",
        "in": "body",
        "required": true,
        "schema": {
            "type": "object",
            "additionalProperties": {
                "format": "int32",
                "type": "integer"
            }
        }
    }

],
```
As you can see, the schema has no "ref", thus there's nowhere for me to put the example.
Furthermore, even if you remove the SwaggerRequestExample attribute from the action method, nothing is displayed in the Swagger-Ui:
![image](https://user-images.githubusercontent.com/169179/29998961-e25bb716-902f-11e7-90f0-3693d6499821.png)

Therefore, this is a problem or limitation of the Swashbuckle library, not mine.

FWIW I tried to figure out a workaround for this for you, e.g.
```
public class PersonRequest2 : Dictionary<int, int>
{
}
```
and then adding that to the action method:
```
[HttpPut]
[Route("api/values/dictionary")]
[SwaggerRequestExample(typeof(DictionaryRequest), typeof(ClicksReturnValueExampleProvider))]
public IHttpActionResult GPutPerson(DictionaryRequest personRequest)
```
but that didn't work. 

There goes my Saturday night.
